airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -0,0 +1,255 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import time
+import math
+import requests
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+
+# from airbyte_cdk.models import SyncMode
+# from airbyte_cdk.sources.streams.http.auth import HttpAuthenticator
+
+
+class ShopifyStream(HttpStream, ABC):
+    url_base = ""
+    primary_key = "id"
+    limit = 250
+
+    def __init__(self, shop: str, api_key: str, api_password: str, api_version: str, **kwargs):
+        super().__init__(**kwargs)
+        self.shop = shop
+        self.api_key = api_key
+        self.api_password = api_password
+        self.api_version = api_version
+        self.since_id = 0
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        decoded_response = response.json()[self.data_field]
+        if len(decoded_response) < self.limit:
+            return None
+        else:
+            self.since_id = decoded_response[-1]["id"]
+            time.sleep(2)
```

Comment:
why is there a sleep? can you leave a comment/

Reply:
This was only for the "slow" debugging" using console output, it's not going to be a part of the code.